rbac/rbac/middleware.py

```diff
@@ -416,6 +417,29 @@ def process_response(self, request, response):  # pylint: disable=no-self-use
         IdentityHeaderMiddleware.log_request(request, response, is_internal)
         return response
 
+    def should_load_user_permissions(self, request: WSGIRequest, user: User) -> bool:
+        """Decide whether RBAC should load the access permissions for the user based on the given request."""
+        # Organization administrators will have already all the permissions so there is no need to load permissions for
+        # them.
+        if user.admin:
+            return False
+
+        # The access endpoint gets a lot of traffic, so we need to restrict for which queries we are actually going
+        # to load the user permissions, since it is a very heavy operation. The following Jira tickets have more
+        # details:
+        #
+        # - RHCLOUD-15394
+        # - RHCLOUD-29631
+        #
+        # There is one use case where we need to load the user's permissions: whenever they want to query for their
+        # or other users' permissions. In that case, we need to know if they're allowed to do so, and for that, we
+        # need to preload their permissions to check them afterwards in the subsequent permission checkers.
+        if request.path.endswith("/access/") and request.method == "GET":
+            query_params: QueryDict = request.GET
+            return "application" in query_params and "username" in query_params
```

Comment:
well it is really tiny comment
"application" is every time in the `query_params` because it is mandatory parameter (it can be empty like `application=` but has to be present in the query) 

so this part of condition is true every time and can be removed
`"application" in query_params`

if you insist that you want this condition there then consider the flip order of conditions into
`"username" in query_params and "application" in query_params`
because python uses lazy evaluation here, when first condition is False then python stop the evaluation (it doesnt matter if the second condition is True or False, the whole expression will be False)